### PR TITLE
Pass Executor into TransportMasterNodeAction

### PR DESCRIPTION
**Pull request by DiannaHohensee**
Related to #97879

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ protected TransportMasterNodeAction(
         Writeable.Reader<Request> request,
         IndexNameExpressionResolver indexNameExpressionResolver,
         Writeable.Reader<Response> response,
-        String executor
+        Executor executor
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java:79]: Serverless has a few callers of this so we need to keep the old API around until we can fix them too.

**Review by DaveCTurner — COMMENTED** [review]
I never realised just how much we used SAME for these actions :(

Anyway looks good except one comment about keeping the old API for a brief period.

**Comment by DiannaHohensee** [comment]
This should be good now. I ran the serverless tests a few days ago and they all passed. I'm running them again now to make certain nothing has changed.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**